base/task.jl

```diff
@@ -326,19 +382,35 @@ end
 
 
 ## dynamically-scoped waiting for multiple items
-
-sync_begin() = task_local_storage(:SPAWNS, ([], get(task_local_storage(), :SPAWNS, ())))
+sync_begin() = task_local_storage(:SPAWNS, (([], CompositeException()), get(task_local_storage(), :SPAWNS, ())))
 
 function sync_end()
     spawns = get(task_local_storage(), :SPAWNS, ())
     if is(spawns,())
         error("sync_end() without sync_begin()")
     end
-    refs = spawns[1]
+    refs = spawns[1][1]
     task_local_storage(:SPAWNS, spawns[2])
+
+    c_ex = spawns[1][2]
```

Comment:
It looks like the CompositeException can be entirely local to `sync_end`; it doesn't need to be stored in SPAWNS.